### PR TITLE
Add removePatientPreferredPharmacy to provider

### PR DESCRIPTION
**Pull request by captDaylight**
In over my skis here, but I need this mutation when setting the preferred pharmacy to remove the old one. 

https://github.com/Photon-Health/app/pull/317

### CODE REVIEW

File: packages/react/src/provider.tsx [hunk 1]
```diff
@@ -1047,6 +1068,66 @@ export const PhotonProvider = (opts: {
...
+    removePatientPreferredPharmacy?: Patient
+    loading: boolean
+    error?: GraphQLError
+  }>({
+    removePatientPreferredPharmacy: undefined,
+    loading: false,
+    error: undefined
+  })
+
+  const removePatientPreferredPharmacyMutation = client.clinical.patient.removePatientPreferredPharmacy({})
```
Comment (jknapp25) [inline packages/react/src/provider.tsx:1082]: ah, so we already have the handler for this, just not serving through the sdk?

Comment (samkotlove) [inline packages/react/src/provider.tsx:1082]: i think this function expect an id (or maybe ids)

**Comment by jknapp25** [comment]
nit: (not on this PR) over in graphql would you mind updating the comments on the removePatientPreferredPharmacy mutation?

- Removes a preferred pharmacy to a patient -> Removes a preferred pharmacy **on** a patient
- ID of the pharmacy to add -> ID of the pharmacy to **remove**

image

File: packages/react/src/provider.tsx (hunk 1 above)
Comment (jknapp25) [reply packages/react/src/provider.tsx:1082]: no, i think it takes `patientId` and `pharmacyId`

Comment (jknapp25) [reply packages/react/src/provider.tsx:1082]: i don't think we're doing anything to handle multiple right now, just storing as an array

Comment (captDaylight) [reply packages/react/src/provider.tsx:1082]: Ya that's my understanding, I'm seeing it in `clinical/patient.ts` but not here in the provider file

https://github.com/Photon-Health/photon-sdk-js/blob/a63daa13a909ad1203f2511fbdf552a4262c9e0e/packages/sdk/src/clinical/patient.ts#L225

Comment (samkotlove) [reply packages/react/src/provider.tsx:1082]: it definitely takes some params

Comment (captDaylight) [reply packages/react/src/provider.tsx:1082]: Yup, but I'm not sure this is where they are passed, I'm looking at similar places where a mutation is declared like  line 1022 `removePatientAllergy`

``` ts
  const removePatientAllergyMutation = client.clinical.patient.removePatientAllergy({})
```

or line 963 for `updatePatient`

``` ts
  const updatePatientMutation = client.clinical.patient.updatePatient({})
```

I changed the variable param to have some context here line 197:

``` ts
  removePatientPreferredPharmacy: ({
    refetchQueries,
    awaitRefetchQueries
  }: {
    refetchQueries: string[]
    awaitRefetchQueries?: boolean
  }) => [
    ({
      variables,
      onCompleted
    }: {
      variables: { patientId: string; pharmacyId: string } // << sup
      onCompleted?: (data: any) => void | undefined
    }) => Promise<void>,
    {
      data: { removePatientPreferredPharmacy: Patient } | undefined | null
      error: GraphQLError
      loading: boolean
    }
  ]
```

but honestly flying a lil blind here until I can test this out somehow...

**Review by cawfeecoder — APPROVED** [review]

**Pull request merged**